trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/ExperimentManagerService.java

```diff
@@ -349,9 +350,15 @@ public Response postExperiment(@RequestBody(content = {
 
         // Instantiate the experiment and return it
         ITmfTrace[] traces = Lists.transform(traceUUIDs, uuid -> TraceManagerService.createTraceInstance(uuid)).toArray(new ITmfTrace[0]);
+        Map<UUID, ITmfTrace> uuidToTraceInstance = new HashMap<>();
+        for (UUID uuid : traceUUIDs) {
+            ITmfTrace trace = TraceManagerService.createTraceInstance(uuid);
+            uuidToTraceInstance.put(uuid, trace);
+        }
+        TRACE_INSTANCES.put(expUUID, uuidToTraceInstance);
```

Comment:
This line should be only be done below when the actual TmfExperiment instantiation was successful (line 378)

Reply:
I didn't realize the traces have a guaranteed order, removed this part.